appengine/bigquery/app.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2015 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+// This App Engine application uses its default service account to list all
+// the BigQuery projects accessible via the BigQuery REST API.
+package sample
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"golang.org/x/net/context"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/bigquery/v2"
+	"google.golang.org/appengine"
+	"google.golang.org/appengine/urlfetch"
+)
+
+func init() {
+	// all requests are handled by handler.
+	http.HandleFunc("/", handle)
+}
+
+func handle(w http.ResponseWriter, r *http.Request) {
+	// create a new App Engine context from the request.
+	c := appengine.NewContext(r)
+
+	// obtain the list of project names.
+	names, err := projects(c)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	// print it to the output.
+	fmt.Fprintln(w, strings.Join(names, "\n"))
+}
+
+// projects returns a list with the names of all the Big Query projects visible
+// with the given context.
```

Comment:
it seems like it'll list all the bigquery projects in the same project as the running App Engine app.